bool has_new_plan
string plan_name
